modular_skyrat/master_files/code/modules/clothing/back/antigravityharness.dm

```diff
@@ -0,0 +1,283 @@
+			if(!user.has_gravity())
+				new /obj/effect/temp_visual/mook_dust/robot(get_turf(src))
+
+			ADD_TRAIT(user, TRAIT_NEGATES_GRAVITY, CLOTHING_TRAIT)
+			playsound(src, 'modular_skyrat/master_files/sound/effects/robot_sit.ogg', 25)
+			to_chat(user, span_notice("Your harness shudders and hisses, projecting a local extra-gravity field."))
+			gravity_on = TRUE
+			icon_state = EXTRAGRAVITY_STATE
+			worn_icon_state = EXTRAGRAVITY_STATE
+
+		if(MODE_GRAVOFF)
+			if(!user.has_gravity() && mode != MODE_GRAVOFF)
+				new /obj/effect/temp_visual/mook_dust/robot(get_turf(src))
+				playsound(src, 'modular_skyrat/master_files/sound/effects/robot_sit.ogg', 25)
+				to_chat(user, span_notice("Your harness lets out a soft whine as your suspension field dissipates, gravity around you normalizing."))
+				mode = MODE_GRAVOFF
+
+			else
+				if(user.has_gravity() && mode != MODE_GRAVOFF)
+					new /obj/effect/temp_visual/mook_dust(get_turf(src))
+					playsound(src, 'sound/effects/gravhit.ogg', 50)
+					to_chat(user, span_notice("Your harness lets out a soft whine as your suspension field dissipates, gravity around you normalizing."))
+					mode = MODE_GRAVOFF
+
+			icon_state = OFF_STATE
+			worn_icon_state = OFF_STATE
+
+		else
+			return FALSE
+
+	update_item_action_buttons()
+	update_appearance()
+
+	return TRUE
+
+
+/obj/item/gravityharness/dropped(mob/user)
+	. = ..()
+	change_mode(MODE_GRAVOFF)
+	user.RemoveElement(/datum/element/forced_gravity, 0)
+	REMOVE_TRAIT(user, TRAIT_NEGATES_GRAVITY, CLOTHING_TRAIT)
+	STOP_PROCESSING(SSobj, src)
+	UnregisterSignal(user, COMSIG_MOB_GET_STATUS_TAB_ITEMS)
+
+/obj/item/gravityharness/attack_self(mob/user)
+	toggle_mode(user, TRUE)
+
+/obj/item/gravityharness/proc/get_status_tab_item(mob/living/source, list/items)
+	SIGNAL_HANDLER
+	items += "Personal Gravitational Field: [mode]"
+	items += "Cell Charge: [current_cell ? "[round(current_cell.percent(), 0.1)]%" : "No Cell!"]"
+
+/obj/item/gravityharness/process(seconds_per_tick)
+	var/mob/living/carbon/human/user = loc
+	if(!user || !ishuman(user) || user.back != src)
+		if(mode != MODE_GRAVOFF)
+			change_mode(MODE_GRAVOFF)
+
+		return
+
+	// Do nothing if the harness isn't emitting gravity of any kind.area
+	if(!gravity_on)
+		return
+
+	// If we got here, the gravity field is on. If there's no cell, turn that shit off
+	if(!current_cell)
+		change_mode(MODE_GRAVOFF)
+
+	// cell.use will return FALSE if charge is lower than GRAVITY_FIELD_COST
+	if(!current_cell.use(GRAVITY_FIELD_COST))
+		to_chat(user, span_warning("The gravitic engine cuts off as [current_cell] runs out of charge."))
+		change_mode(MODE_GRAVOFF)
+
+/obj/item/gravityharness/Destroy()
+	if(isatom(current_cell))
+		QDEL_NULL(current_cell)
+
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/gravityharness/get_cell()
+	if(cell_cover_open)
+		return current_cell
+
+/obj/item/gravityharness/handle_atom_del(atom/harnesscell)
+	if(harnesscell == current_cell)
+		change_mode(MODE_GRAVOFF)
```

Comment:
You also probably want to do `current_cell = null` here after the `change_mode` call, so we avoid hard-deletions.